maven-plugin/src/main/java/org/openmrs/maven/plugins/utility/DefaultWizard.java

```diff
@@ -270,6 +277,137 @@ public String promptForPlatformVersion(List<String> versions) {
         return version;
 	}
 
+    @Override
+    public String promptForJdkPath(Server server) {
+
+        List<String> paths = new ArrayList<>();
+        paths.add("JAVA_HOME (currently: " + System.getProperty("java.home") + ")");
+        try {
+            paths.addAll(getJdkPathsFromProperties(getSdkProperties()));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        if (!interactiveMode) {
+            return paths.get(0);
+        }
+
+        String path = promptForMissingValueWithOptions(SDKConstants.OPENMRS_SDK_JDK_OPTION,
+                server.getJdkPath(), "path", paths, SDKConstants.OPENMRS_SDK_JDK_CUSTOM, null);
+
+        if (path.equals(paths.get(0))) {
+            server.setJdkPath(System.getProperty("java.home"));
+        }
+
+        if (!isThereJdkUnderPath(path)) {
+                System.out.println(SDKConstants.OPENMRS_SDK_JDK_CUSTOM_INVALID);
+                server.setJdkPath(promptForJdkPath(server));
+        }
+        else {
+            if (!isPathLineDuplicated(paths, path)) {
+                try {
+                    addJdkPathToSdkProperties(path);
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+            return path;
+        }
+        return null;
+    }
+
+    private void addJdkPathToSdkProperties(String path) throws IOException {
```

Comment:
It's better to catch IOException and rethrow as IllegalStateException (RuntimeException) inside this method as there's nothing one can do when calling it.